Assets/Scripts/Game/Questing/Quest.cs

```diff
@@ -339,21 +347,6 @@ void ClearPendingClickRearms()
             pendingClickRearms.Clear();
         }
 
-        /// <summary>
-        /// initializes quest rumors
-        /// dictionary must contain the quest's messages (call after messages was initialized correctly)
-        /// </summary>
-        public void initQuestRumors()
-        {
-            // Add RumorsDuringQuest rumor to rumor mill
-            Message message;
-            message = GetMessage((int)QuestMachine.QuestMessages.RumorsDuringQuest);
-            if (message != null)
```

Comment:
Please note the null check here. Your replacement doesn't preserve this and causes a crash at main quest startup. I added more detailed notes in general comments on this PR.

Reply:
My bad.
Should `AddOrReplaceQuestProgressRumor` do an argument null check and throw an ArgumentNullException too?
I noticed that public methods by TheLacus tend to check their arguments but other code does not.

Reply:
Yeah, it certainly can't hurt. I'm just doing some final checks and will merge this today if no other issues found. I'll pop in that null reference check before merging.